src/os_auth/main-server.c

```diff
@@ -1035,9 +1035,9 @@ void* run_dispatcher(__attribute__((unused)) void *arg) {
 
             /* Check for agents limit */
 
-            if (config.flags.register_limit && keys.keysize >= (MAX_AGENTS - 2) ) {
+            if (config.flags.register_limit && keys.keysize >= (MAX_AGENTS - 3) ) {
```

Comment:
Hi, could you avoid magic numbers?
keys.keysize is const? if it is not, you could put the definition macro to the left of the condition to avoid assignments in typo errors.